File: JAVA/baseDatos/Serializador.java
package baseDatos;

import gestionApp.Empresa;
import gestionApp.Evento.*;
import gestionApp.Negocio;
import gestionApp.Venta;
import gestionApp.personas.*;
import gestionApp.servicios.*;

import java.io.*;
import java.util.ArrayList;

public class Serializador {
    public static <E> void serializar(ArrayList<E> lista, String clase) {
        FileOutputStream fileOut;
        try {
            String path = System.getProperty("user.dir") + "//baseDatos//temp//" + clase + ".txt";
            fileOut = new FileOutputStream(path);
            ObjectOutputStream out = new ObjectOutputStream(fileOut);
            out.writeObject(lista);
            out.close();
            fileOut.close();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void serializacion() {
        serializar(Cliente.getAllClientes(), "Cliente");
        serializar(Empleado.getAllEmpleados(), "Empleado");
        serializar(Empresa.getAllEmpresas(), "Empresa");
        serializar(Negocio.getNegocios(), "Negocio");
        serializar(Venta.getAllVenta(), "Venta");
        serializar(Promocion.getAllPromocion(), "Promocion");
        serializar(Servicio.getAllServicios(), "Servicio");
        serializar(Evento.getAllEventos(), "Evento");



    }

}

File: Python/baseDatos/serializador.py
import pickle
import pathlib
import os

from Python.gestionApp.personas.Empleado import Empleado
from Python.gestionApp.personas.Cliente import Cliente


class Serializador:

    @classmethod
    def serializar(cls):
        datos = {
            "empleado": Empleado.getAllEmpleados(),
            "cliente": Cliente.getAllClientes()
        }

        for archivo, dato in datos.items():
            picklefile = open(os.path.join(pathlib.Path(__file__).parent.absolute(), f"temp\\{archivo}"), "wb")
            pickle.dump(dato, picklefile)
            picklefile.close()
